src/modules/todos/controllers.js

```diff
@@ -1,144 +1,154 @@
+			.replace("M", today.getMinutes())
+			.replace("s", today.getSeconds());
+		const todo_id = v4();
+		const user_id = req.userId;
+		const { title, status, dueDate } = req.body;
 		const createdAt = strDate;
 		const updatedAt = strDate;
 		const todo = {
-			id,
-			userId,
+			todo_id,
+			user_id,
 			title,
 			status,
 			dueDate,
 			createdAt,
 			updatedAt,
-		}
-		todos.push(todo);
+		};
+		const newTodo = await pool.query(queries.newTodo, [
+			todo.todo_id,
+			todo.user_id,
+			todo.title,
+			todo.status,
+			todo.dueDate,
+			todo.createdAt,
+			todo.updatedAt,
+		]);
 		return res.status(201).json({
-			statcode: 201,
-			data: {
-				id: todo.id,
-				userId: todo.userId,
-				title: todo.title,
-				status: todo.status,
-				dueDate: todo.dueDate,
-				createdAt: todo.createdAt,
-				updatedAt: todo.updatedAt,
-			}
-		})
+			statuscode: 201,
+			data: newTodo.rows[0],
+		});
 	} catch (err) {
 		next(err);
-	};
-};
-
-export function getTodo(req, res) {
-	const id = req.params.id;
-	const findTodo = todos.find((todo) => todo.id === id);
-	if (!findTodo) {
-		throw new AppException(404, "Unable to retrieve todo")
 	}
-	else if (findTodo.userId === req.userId) {
-		return res.status(200).json({
-			statusCode: 200,
-			data: {
-				id: findTodo.id,
-				userId: findTodo.userId,
-				title: findTodo.title,
-				status: findTodo.status,
-				dueDate: findTodo.dueDate,
-				createdAt: findTodo.createdAt,
-				updatedAt: findTodo.updatedAt,
-			},
-		});
-	}
-	else {
-		throw new AppException(403, "Unauthorized")
+}
+
+export async function getTodo(req, res, next) {
+	try {
+		const { id } = req.params;
+		const findTodo = await pool.query(queries.findTodo, [id]);
+		if (!findTodo.rowCount) {
+			throw new AppException(404, "Unable to retrieve todo");
+		} else if (findTodo.rows[0].user_id === req.userId) {
+			return res.status(200).json({
+				statusCode: 200,
+				data: findTodo.rows[0],
+			});
+		} else {
+			throw new AppException(403, "Unauthorized");
+		}
+	} catch (err) {
+		next(err);
 	}
-};
+}
 
-export function updateTodo(req, res) {
-	let today = new Date();
-		
-	let strDate = 'Y-m-d h:M:s'
-		.replace('Y', today.getFullYear())
-		.replace('m', today.getMonth()+1)
-		.replace('d', today.getDate())
-		.replace('h', today.getHours())
-		.replace('M', today.getMinutes())
-		.replace('s', today.getSeconds());
+export async function updateTodo(req, res, next) {
+	try {
+		let today = new Date();
 
-	const id = req.params.id;
-	const title = req.body.title;
-	const status = req.body.status
-	const dueDate = req.body.dueDate;
-	const updateTodo = todos.find((todo) => todo.id === id);
-	if (!updateTodo) {
-		throw new AppException(404, "Unable to retrieve todo")
-	}
-	else if (updateTodo.userId === req.userId) {
-		if (title) {
-			updateTodo.title = title;
-		}
-		if (status) {
-			updateTodo.status = status;
-		}
-		if (dueDate) {
-			updateTodo.dueDate = dueDate;
+		let strDate = "Y-m-d h:M:s"
```

Comment:
Again this string replace will no longer be needed if you use the correct data type for the date columns in the DB.